docs/setup/keadm.md

```diff
@@ -96,70 +96,229 @@ KubeEdge edgecore is running, For logs visit:  /var/log/kubeedge/edgecore.log
-cd metrics-server
-make container
-docker images
-docker tag 2d9eb6e0d887 metrics-server-kubeedge:latest
-```
+    Git clone latest metrics server repository:
+
+    ```bash
+    git clone https://github.com/kubernetes-sigs/metrics-server.git
+    ```
+   
+    Go to the metrics server directory:
+
+    ```bash
+    cd metrics-server
+    ```
+
+    Make the docker image:
+
+    ```bash
+    make container
+    ```
+
+    Check if you have this docker image:
+
+    ```bash
+    docker images
+    ```
+
+    |                  REPOSITORY                           |                    TAG                   |   IMAGE ID   |     CREATE     |  SIZE  |
+    |-------------------------------------------------------|------------------------------------------|--------------|----------------|--------|
+    | gcr.io/k8s-staging-metrics-serer/ metrics-serer-amd64 | 6d92704c5a68cd29a7a81bce68e6c2230c7a6912 | a24f71249d69 | 19 seconds ago | 57.2MB |
+    | metrics-server-kubeedge                               |                 latest                   | aef0fa7a834c | 28 seconds ago | 57.2MB |
+
+
+    Make sure you change the tag of image by using its IMAGE ID to be compactable with image name in yaml file.
+
+    ```bash
+    docker tag a24f71249d69 metrics-server-kubeedge:latest
+    ```
 
 3. Apply the deployment yaml. For specific deployment documents, you can refer to https://github.com/kubernetes-sigs/metrics-server/tree/master/manifests.
 
+    Before you deploy metrics-server, you have to make sure that you deploy it on the node which has apiserver deployed on. In this case, that is the master node. As a consequence, it is needed to make master node schedulable by the following command:
+
+    ``` shell
+    kubectl taint nodes --all node-role.kubernetes.io/master-
+    ```
+   
+    Then, in the deployment.yaml file, it must be specified that metrics-server is deployed on master node.
+    (The hostname is chosen as the marked label.)
+    In **metrics-server-deployment.yaml**
+    ``` yaml
+        spec:
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  #Specify which label in [kubectl get nodes --show-labels] you want to match
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    #Specify the value in key
+                    - charlie-latest
+    ```
+
 **IMPORTANT NOTE:**
 1. Metrics-server needs to use hostnetwork network mode.
 
-2. Use the image compiled by yourself and set imagePullPolicy to IfNotPresent.
+2. Use the image compiled by yourself and set imagePullPolicy to Never.
 
 3. Enable the feature of --kubelet-use-node-status-port for Metrics-server
 
+    Those settings need to be written in deployment yaml (metrics-server-deployment.yaml) file like this:
+
+    ``` yaml
```

Comment:
maybe we can offer an entire deployment yaml?

Reply:
I did. But @Poor12 said that the original one is good enough yesterday, the one I used is here.

Ref:
 The deployment YAML files can be found in this repository:
    [DockerAndKubernetesCourseCode](https://github.com/DanWahlin/DockerAndKubernetesCourseCode/tree/master/samples/prometheus/metrics-server)

![image](https://user-images.githubusercontent.com/40239192/91865200-b80e7100-eca3-11ea-9ed2-770427391e82.png)

Reply:
@luogangyi I have discussed this issue with fisher and daixiang. For kubeedge users, ms is not necessary and there is no need to write yaml files directly in kubeedge.